test/index.test.ts

```diff
@@ -1,6 +1,8 @@
 import { expect, use } from 'chai';
 import chaiAsPromised from 'chai-as-promised';
-import 'mocha';
+
```

Comment:
i did this to not import context from @types/mocha folder
tests still works without problems

Reply:
still don't get it - this is an entirely different file!  This file should NOT have an affect on the other file.

Okay I investigated a little myself...  I think there's a combination of problems here we need to solve:
1. Global namespace pollution shouldn't be allowed at all!  Let alone, pollution ACROSS FILES!  imports like this shouldn't be allowed at all!  After a quick search I didn't find a tsconfig way to make it illegal, but I found tslint's "no-import-side-effect".  I'm not sure if it's exactly what we want though.
- [ ] FUI to make it illegal and verify our solution solves the problem
2. We should be making circle confirm that we can do a successful build through npm build too
- [ ] FUI to update CI config so "npm build" failure is needed too.
3. tsconfig.json is used by vscode - but that's our TEST tsconfig.  Instead, that file should be renamed to tsconfig.base.json, and tsconfig.build.json should be renamed to tsconfig.json, and we need a new tsconfig.test.json (which test should use).
- [ ] FUI to reorganize, fix, and clean up tsconfig stuff
4. We should avoid the name "context" just in case from now on - htCtx instead maybe.
- [ ] FUI to rename generic names like "context" everywhere

Reply:
https://github.com/captaincaius/hipthrusts/issues/21#issuecomment-617110824